src/AdaptiveCardContainer.tsx

```diff
@@ -0,0 +1,208 @@
+import * as React from 'react';
+import * as AdaptiveCards from "microsoft-adaptivecards";
+import { CardAction } from "botframework-directlinejs/built/directLine";
+import { classList, IDoCardAction, konsole } from "./Chat";
+import { AjaxResponse, AjaxRequest } from 'rxjs/observable/dom/AjaxObservable';
+import { Observable } from 'rxjs/Observable';
+
+export interface Props {
+    card: any,
+    onImageLoad?: () => any,
+    onClick?: (e: React.MouseEvent<HTMLElement>) => void,
+    onCardAction: IDoCardAction,
+    className?: string
+}
+
+class LinkedAdaptiveCard extends AdaptiveCards.AdaptiveCard {
+    constructor(public adaptiveCardContainer: AdaptiveCardContainer) {
+        super();
+    }
+}
+
+export interface BotFrameworkCardAction extends CardAction {
+    __isBotFrameworkCardAction: boolean
+}
+
+function getLinkedAdaptiveCard(action: AdaptiveCards.Action) {
+    let element = action.parent;
+    while (element && !(element instanceof LinkedAdaptiveCard)) {
+        element = element.parent;
+    }
+    return element as LinkedAdaptiveCard;
+}
+
+AdaptiveCards.AdaptiveCard.onExecuteAction = (action: AdaptiveCards.ExternalAction) => {
+
+    if (action instanceof AdaptiveCards.OpenUrlAction) {
+        window.open(action.url);
+
+    } else if (action instanceof AdaptiveCards.SubmitAction) {
+        const linkedAdaptiveCard = getLinkedAdaptiveCard(action);
+        if (linkedAdaptiveCard && action.data) {
+            if (typeof action.data === 'object' && (action.data as BotFrameworkCardAction).__isBotFrameworkCardAction) {
+                const cardAction = (action.data as BotFrameworkCardAction);
+                linkedAdaptiveCard.adaptiveCardContainer.onCardAction(cardAction.type, cardAction.value);
+            } else {
+                linkedAdaptiveCard.adaptiveCardContainer.onCardAction('postBack', action.data);
+            }
+        }
+
+    } else if (action instanceof AdaptiveCards.HttpAction) {
+        /**
```

Comment:
If we're not going to support HTTP Actions to start we should simply ignore the buttons, UI, etc.